### PR TITLE
Allow calling constants from contracts as well

### PR DESCRIPTION
**Pull request by sushi-shi**
Signed-off-by: Iwan Waitusenok <sleepplease@protonmail.ch>

### CODE REVIEW

File: src/sema/expression.rs [hunk 1]
```diff
@@ -4907,8 +4907,8 @@ fn member_access(
         return Ok(expr);
     }
 
-    // is it a constant in a library (unless basecontract is a local variable)
-    match library_constant(e, id, file_no, ns, symtable) {
+    // is it a constant (unless basecontract is a local variable)
+    match constant(e, id, file_no, ns, symtable) {
         None => (),
         Some(expr) => return Ok(expr),
```
Comment (seanyoung) [inline src/sema/expression.rs:4913]: This seems like an baroque way of saying:

```
if let Some(expr) =  constant(e, id, file_no, ns, symtable) {
    return Ok(expr);
}
```

What do yo think?

File: src/sema/expression.rs [hunk 2]
```diff
@@ -5251,7 +5251,7 @@ fn member_access(
     Err(())
 }
 
-fn library_constant(
+fn constant(
```
Comment (seanyoung) [inline src/sema/expression.rs:5254]: `fn constant()` is not a great name. How about `fn contract_constant()`?

File: src/sema/expression.rs [hunk 3]
```diff
@@ -5269,16 +5269,8 @@ fn library_constant(
 
     let library_no = ns.resolve_contract(file_no, namespace)?;
```
Comment (seanyoung) [inline src/sema/expression.rs:5270]: contract_no?

File: src/sema/expression.rs (hunk 1 above)
Comment (sushi-shi) [reply src/sema/expression.rs]: Yes, of course!

**Comment by seanyoung** [comment]
Very nice :+1:

**Pull request merged**